File: bench/node/base64.js
const N = parseInt(process.env.BENCH_N || "50000", 10);
let x = 123456789;
const bytes = Buffer.alloc(N);
for (let i = 0; i < N; i++) { x = (Math.imul(x, 1103515245) + 12345) & 0x7FFFFFFF; bytes[i] = x % 256; }
const enc = bytes.toString("base64");
const dec = Buffer.from(enc, "base64");
let encSum = 0;
for (let i = 0; i < enc.length; i++) encSum = (encSum + enc.charCodeAt(i)) % 2147483647;
let decSum = 0;
for (let i = 0; i < dec.length; i++) decSum = (decSum + dec[i]) % 2147483647;
console.log((encSum + decSum) % 2147483647);

File: bench/go/base64.go
// base64 encode+decode with the standard library's encoding/base64.
package main

import (
	"encoding/base64"
	"fmt"
)

func main() {
	n := benchN(50000)
	x := 123456789
	bytes := make([]byte, n)
	for i := 0; i < n; i++ {
		x = (x*1103515245 + 12345) & 0x7FFFFFFF
		bytes[i] = byte(x % 256)
	}
	enc := base64.StdEncoding.EncodeToString(bytes)
	dec, err := base64.StdEncoding.DecodeString(enc)
	if err != nil {
		panic(err)
	}
	encSum := 0
	for i := 0; i < len(enc); i++ {
		encSum = (encSum + int(enc[i])) % 2147483647
	}
	decSum := 0
	for _, b := range dec {
		decSum = (decSum + int(b)) % 2147483647
	}
	fmt.Println((encSum + decSum) % 2147483647)
}
